src/main/java/bookstore/repository/book/spec/TitleSpecificationProvider.java

```diff
@@ -0,0 +1,20 @@
+package bookstore.repository.book.spec;
+
+import bookstore.model.Book;
+import bookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TitleSpecificationProvider implements SpecificationProvider<Book> {
```

Comment:
let's use "like" logic for Title
for example when we have "master" in parameter we would like to see both
**Master and Margarita** and **Master Your Emotions** in the results